// @edx/frontend-lib-learning-assistant/dist/components/ChatBox/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _react = _interopRequireDefault(require("react"));
var _reactRedux = require("react-redux");
var _propTypes = _interopRequireDefault(require("prop-types"));
var _Message = _interopRequireDefault(require("../Message"));
require("./ChatBox.scss");
var _jsxRuntime = require("react/jsx-runtime");
function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }
// container for all of the messages
const ChatBox = _ref => {
  let {
    chatboxContainerRef
  } = _ref;
  const {
    messageList,
    apiIsLoading
  } = (0, _reactRedux.useSelector)(state => state.learningAssistant);
  return /*#__PURE__*/(0, _jsxRuntime.jsxs)("div", {
    ref: chatboxContainerRef,
    className: "flex-grow-1 scroller d-flex flex-column pb-4",
    children: [messageList.map(_ref2 => {
      let {
        role,
        content,
        timestamp
      } = _ref2;
      return /*#__PURE__*/(0, _jsxRuntime.jsx)(_Message.default, {
        variant: role,
        message: content
      }, timestamp.toString());
    }), apiIsLoading && /*#__PURE__*/(0, _jsxRuntime.jsx)("div", {
      className: "loading",
      children: "Xpert is thinking"
    })]
  });
};
ChatBox.propTypes = {
  chatboxContainerRef: _propTypes.default.oneOfType([_propTypes.default.func, _propTypes.default.shape({
    current: _propTypes.default.instanceOf(Element)
  })]).isRequired
};
var _default = ChatBox;
exports.default = _default;
